tests/components/verisure/test_config_flow.py

```diff
@@ -0,0 +1,357 @@
+        await hass.async_block_till_done()
+
+    assert result3["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result3["title"] == "descending (54321th street)"
+    assert result3["data"] == {
+        CONF_GIID: "54321",
+        CONF_EMAIL: "verisure_my_pages@example.com",
+        CONF_PASSWORD: "SuperS3cr3t!",
+    }
+
+    assert len(mock_verisure.mock_calls) == 2
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_invalid_login(hass: HomeAssistant) -> None:
+    """Test a flow with an invalid Verisure My Pages login."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.verisure.config_flow.Verisure.login",
+        side_effect=VerisureLoginError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "email": "verisure_my_pages@example.com",
+                "password": "SuperS3cr3t!",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "user"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_unknown_error(hass: HomeAssistant) -> None:
+    """Test a flow with an invalid Verisure My Pages login."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.verisure.config_flow.Verisure.login",
+        side_effect=VerisureError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "email": "verisure_my_pages@example.com",
+                "password": "SuperS3cr3t!",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "user"
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_dhcp(hass):
+    """Test that DHCP discovery works."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        data={MAC_ADDRESS: "01:23:45:67:89:ab"},
+        context={"source": config_entries.SOURCE_DHCP},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_options_flow(hass):
+    """Test options config flow."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="12345",
+        data={},
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "init"
+    schema = result["data_schema"].schema
+    assert (
+        _get_schema_default(schema, CONF_LOCK_CODE_DIGITS) == DEFAULT_LOCK_CODE_DIGITS
```

Comment:
Maybe we can just call the schema with some input, eg empty dict, and check that defaults are put in the returned value?